lib/ClangImporter/ClangModuleDependencyScanner.cpp

```diff
@@ -156,54 +137,7 @@ ModuleDependencyVector ClangImporter::bridgeClangModuleDependencies(
     }
 
     // Add args reported by the scanner.
-
-    // Round-trip clang args to canonicalize and clear the options that swift
-    // compiler doesn't need.
-    clang::CompilerInvocation depsInvocation;
-    clang::DiagnosticsEngine clangDiags(new clang::DiagnosticIDs(),
-                                        new clang::DiagnosticOptions(),
-                                        new clang::IgnoringDiagConsumer());
-
-    llvm::SmallVector<const char*> clangArgs;
-    llvm::for_each(
-        clangModuleDep.getBuildArguments(),
-        [&](const std::string &Arg) { clangArgs.push_back(Arg.c_str()); });
-
-    bool success = clang::CompilerInvocation::CreateFromArgs(
-        depsInvocation, clangArgs, clangDiags);
-    (void)success;
-    assert(success && "clang option from dep scanner round trip failed");
-
-    // Create a prefix mapper that matches clang's configuration.
-    auto Mapper =
-        getClangPrefixMapper(clangScanningTool, clangModuleDep, depsInvocation);
-
-    // Clear the cache key for module. The module key is computed from clang
-    // invocation, not swift invocation.
-    depsInvocation.getFrontendOpts().ModuleCacheKeys.clear();
-    depsInvocation.getFrontendOpts().PathPrefixMappings.clear();
-    depsInvocation.getFrontendOpts().OutputFile.clear();
-
-    // Reset CASOptions since that should be coming from swift.
-    depsInvocation.getCASOpts() = clang::CASOptions();
-    depsInvocation.getFrontendOpts().CASIncludeTreeID.clear();
-
-    // FIXME: workaround for rdar://105684525: find the -ivfsoverlay option
```

Comment:
Note this is dropped. I don't think we need this anymore but might need some testing to make sure.

Reply:
This is actually needed for non-caching build. I improve the tests to actually error out when the state of error triggers. Brought it back with new edition of API.